app/src/main/java/org/android/go/sopt/presentation/auth/SignUpActivity.kt

```diff
@@ -50,56 +39,56 @@ class SignUpActivity : AppCompatActivity() {
             hideKeyboard(this)
         }
 
-        // 받아온 클래스 활용해 스크롤뷰에 적용
-        keyboardVisibilityUtils = KeyboardVisibilityUtils(window,
-            onShowKeyboard = { keyboardHeight ->
-                binding.svSignUp.run {
-                    smoothScrollTo(scrollX, scrollY + keyboardHeight)
-                }
-            })
+        // 키보드 높이만큼 EditText 올려 버튼이 가리지 않도록 설정
+        setKeyboardHeight()
+    }
+
+    private fun observeSignUpFormat() {
+        viewModel.isIdValid.observe(this) { isIdValid ->
+            binding.layoutSignUpId.error = if (isIdValid) "" else "아이디 형식이 올바르지 않습니다."
+            viewModel.checkButtonValid()
+        }
+        viewModel.isPwValid.observe(this) { isPwValid ->
+            binding.layoutSignUpPw.error = if (isPwValid) "" else "비밀번호 형식이 올바르지 않습니다."
+            viewModel.checkButtonValid()
+        }
+    }
+
+    private fun observeSignUpResult() {
+        viewModel.signUpResult.observe(this) { signUpResult ->
+            binding.root.makeSnackBar(getString(R.string.snackbar_signup_success))
+            if (!isFinishing) {
+                val intent = Intent(binding.root.context, LoginActivity::class.java)
+                startActivity(intent)
+                finish()
+            }
+        }
+        viewModel.errorResult.observe(this) { errorResult ->
+            Timber.d("서버 통신 실패 : $errorResult")
+            binding.root.makeSnackBar(getString(R.string.snackbar_server_failure))
+        }
+    }
+
+    private fun signUpWithServer() {
```

Comment:
바인딩 변수로 써주는 것보다는 현재 양방향 데이터 바인딩을 사용하고 계시니 뷰모델 변수를 호출해서 사용하는게 추후 유지보수하기에도 좋을 것 같습니다 !
```suggestion
private fun signUpWithServer() {
        viewModel.signUp(
             viewModel. idText.value.toString(),
            viewModel. pwText.value.toString(),
            viewModel. nameText.value.toString(),
            viewModel. skillText.value.toString()
        )
    }
```

Reply:
아핫 수정해둘게요 !

Reply:
아래에서 signUp 자체를 수정해서 아예 삭제했습니당